api-gateway/src/main/kotlin/com/saveourtool/save/gateway/service/BackendService.kt

```diff
@@ -52,6 +55,38 @@ class BackendService(
             responseEntity.body.toMono().orNotFound { "Authentication body is empty" }
         }
 
+    /**
+     * Find current user [SaveUserDetails] by [authentication].
+     *
+     * @param authentication current user [Authentication]
+     * @return current user [SaveUserDetails]
+     */
+    fun findByAuthentication(authentication: Authentication): Mono<SaveUserDetails> = when (authentication) {
+        is UsernamePasswordAuthenticationToken -> findByName(authentication.name)
+        is OAuth2AuthenticationToken -> {
+            val source = authentication.authorizedClientRegistrationId
```

Comment:
I would actually move this code to some common  method...

Reply:
the point is that `UsernamePasswordAuthenticationToken` path varies - sometimes we want to make a request while sometimes we don't want

Reply:
and btw, we have similar logic here `AuthorizationHeadersGatewayFilterFactory`